lib/tests/test_merged_tree.rs

```diff
@@ -1182,3 +1183,153 @@ fn test_merge_simplify_result() {
     let merged = side1_merged.merge(&base1_merged, &side2_merged).unwrap();
     assert_eq!(merged, expected_merged);
 }
+
+/// Test that we simplify content-level conflicts before passing them to
+/// files::merge().
+///
+/// This is what happens when you squash a conflict resolution into a conflict
+/// and it gets propagated to a child where the conflict is different.
+#[test]
+fn test_merge_simplify_file_conflict() {
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    let conflict_path = RepoPath::from_internal_string("CHANGELOG.md");
+    let other_path = RepoPath::from_internal_string("other");
+
+    let text1 = r#"### New features
+
+* The `ancestors()` revset function now takes an optional `depth` argument
```

Comment:
Are the first three paragraphs (including `### New features`) the same every time? Would the test still work if you removed one or both of them?

If they are the same but cannot be removed, could the common part be extracted to a separate variable? Then, you could have

```
let text1 = format!("{common_part}

...
"
```

(Or we could use `fromatdoc!` form `indoc`, though it wouldn't matter much here).

Reply:
I actually tried that before but gave up before I finished. I like the result now that it's done :)